fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/observer/SimpleObserver.java

```diff
@@ -68,6 +82,60 @@
     static final Integer EVENT_TYPES = NODE_ADDED + NODE_REMOVED + NODE_MOVED + PROPERTY_ADDED + PROPERTY_CHANGED
             + PROPERTY_REMOVED;
 
+    private static Stream<FedoraResource> getAllChildren(final FedoraResource resource) {
+        return concat(of(resource), resource.getChildren().flatMap(SimpleObserver::getAllChildren));
+    }
+
+    /**
+     * Note: In order to resolve the JCR-based properties (e.g. jcr:lastModified or
+     * ebucore:mimeType) a new FedoarEvent is created and the dereferenced properties
+     * are added to that new event.
+     */
+    private static Function<Session, Function<FedoraEvent, FedoraEvent>> namespaceResolver = session -> evt -> {
+        final NamespaceRegistry namespaceRegistry = getNamespaceRegistry(session);
+        final FedoraEvent event = new FedoraEventImpl(evt.getTypes(), evt.getPath(), evt.getUserID(),
+                evt.getUserData(), evt.getDate(), evt.getInfo());
+
+        evt.getProperties().stream()
+            .filter(isPublicJcrProperty.or(property -> !property.startsWith("jcr:")))
+            .forEach(property -> {
+                final String[] parts = property.split(":", 2);
+                if (parts.length == 2) {
+                    try {
+                        event.addProperty(
+                            getRDFNamespaceForJcrNamespace(namespaceRegistry.getURI(parts[0])) + parts[1]);
+                    } catch (final RepositoryException ex) {
+                        LOGGER.warn("Prefix could not be dereferenced using the namespace registry, skipping: {}",
+                            property);
+                    }
+                } else {
+                    LOGGER.warn("Prefix could not be determined, skipping: {}", property);
+                }
+            });
+        return event;
+    };
+
+    private static Function<Session, Function<FedoraEvent, Stream<FedoraEvent>>> handleMoveEvents = session -> evt -> {
```

Comment:
Can we have a method-level doc explaining the semantics of this function? I understand what it does-- I want to doc what it is _supposed_ to do.